src/app/address/controllers/address/nonUKAddress.js

```diff
@@ -0,0 +1,124 @@
+const BaseController = require("hmpo-form-wizard").Controller;
+const countryList = require("../../data/countries.json");
+const {
+  validateBuildingAddress,
+} = require("../../validators/nonUKAddressValidator");
+
+class NonUKAddressController extends BaseController {
+  getValues(req, res, callback) {
+    super.getValues(req, res, (err, values) => {
+      values.addressCountryName = this.mapToName(
+        req.translate,
+        req.sessionModel.get("country")
+      );
+
+      Object.entries(req?.form?.errors || {})?.map(
+        ([fieldName, validationAttribute]) =>
+          (values[`${fieldName}InValid`] = this.getErrorMessage(
+            req.translate,
+            validationAttribute.key,
+            validationAttribute.type
+          ))
+      );
+
+      values.buildAddressErrorMessage = this.hasBuildingEmptyErrors(req) && {
+        text: req.translate("validation.apartmentOrBuildingNumberOrName"),
+        visuallyHiddenText: "error",
+      };
+
+      callback(err, values);
+    });
+  }
+
+  validateFields(req, res, callback) {
+    const formFields = req.form.options.fields;
+    const apartment = "nonUKAddressApartmentNumber";
+    const number = "nonUKAddressBuildingNumber";
+    const name = "nonUKAddressBuildingName";
+
+    const fields = {
+      [apartment]: req.body.nonUKAddressApartmentNumber,
```

Comment:
[nit] could you reuse the constants again? Something like `[apartment]: req.body[apartment]`?